app/(app)/offence/index.tsx

```diff
@@ -49,39 +55,53 @@ const OffencePage = () => {
     [role?.actions.zone, offenceType, location, udrData],
   )
 
-  const onSubmit = async () => {
-    if (isSubmitting) {
-      return
-    }
-
-    setIsSubmitting(true)
-    setIsTouched(true)
-
-    if (!(offenceType && (isObjectiveResponsibility || resolutionType) && ecv) || isLocationError) {
-      setIsSubmitting(false)
-
-      return
-    }
-
-    const response = await clientApi.scanControllerGetOffenceList(
-      ecv,
-      undefined,
-      undefined,
-      location?.lat,
-      location?.long,
-    )
-
-    if (response.data.length > 0) {
-      openModal()
-      setIsSubmitting(false)
-
-      return
-    }
-
-    router.navigate('/offence/vehicle')
-    setIsSubmitting(false)
-    setIsTouched(false)
-  }
+  const { mutate, isPending, isError } = useMutation({
+    mutationFn: async () => {
+      if (
+        !(offenceType && (isObjectiveResponsibility || resolutionType) && ecv && scanData) ||
```

Comment:
It's not really clear for me, why we check for `offenceType`, `isObjectiveResponsibility` or `resolutionType` if they are not used in the endpoints. Maybe add a comment please?